src/main/java/editorconfig/JGitResource.java

```diff
@@ -0,0 +1,134 @@
+package editorconfig;
+
+import org.ec4j.core.Resource;
+import org.ec4j.core.ResourcePath;
+import org.ec4j.core.model.Ec4jPath;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.lib.ObjectId;
+import org.eclipse.jgit.lib.ObjectReader;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.revwalk.RevTree;
+import org.eclipse.jgit.revwalk.RevWalk;
+import org.eclipse.jgit.treewalk.TreeWalk;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.charset.StandardCharsets;
+
+public class JGitResource implements Resource {
```

Comment:
`JGitResource` and `JGitResourcePath` can be implemented with Scala. How about implementing them as inner classes of `EditorConfigUtil`?